extensions/vi-mode/commands.lisp

```diff
@@ -689,14 +705,23 @@ Move the cursor to the first non-blank character of the line."
 (defun vi-forward-matching-paren (window point &optional (offset -1))
   (declare (ignore window))
   (with-point ((point point))
+    (loop :until (or (syntax-open-paren-char-p (character-at point))
+                     (syntax-closed-paren-char-p (character-at point))
+                     (= (point-charpos point) (length (line-string point))))
+          :do (incf (point-charpos point)))
```

Comment:
Oh, I realized after merging that changing point-charpos breaks encapsulation.
I think it would be better to use line-end-p and character-offset instead.

Reply:
Ok, I'll do it within 24h. Thanks!

Reply:
Would it still a problem with temporary point? Like `(with-point ((point (copy-point point))) ...)`

Reply:
yes, but copy-point is not necessary because it is used automatically with-point.
`(with-point ((point point)) ...)`

Reply:
`(apropos "line-end-p")` doesn't find anything. There's a struct `line-end-item-p` but I don't think it's what I need here.

Reply:
Oh, you mean `(line-end p)`. :)

Reply:
Sorry, it was end-line-p.